### PR TITLE
feat: Added `enableLookAhead` option to relations

### PR DESCRIPTION
**Pull request by TriPSs**
Todo:
- [x] Tests
- [x] Docs

### CODE REVIEW

File: packages/query-graphql/src/decorators/graphql-resolve-info.utils.ts [hunk 1]
```diff
@@ -0,0 +1,98 @@
+import { Kind } from 'graphql'
```
Comment (TriPSs) [inline packages/query-graphql/src/decorators/graphql-resolve-info.utils.ts:1]: This file needs cleaned up.

File: packages/query-graphql/src/decorators/graphql-resolve-info.utils.ts [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+
+import type { CursorConnectionType, OffsetConnectionType } from '../types'
+import type { Query } from '@ptc-org/nestjs-query-core'
+import type { GraphQLResolveInfo as ResolveInfo } from 'graphql'
+
+export interface QueryResolveTree<DTO> {
+  name: string
+  alias: string
+  args?: Query<DTO>
+  fieldsByTypeName: {
```
Comment (TriPSs) [inline packages/query-graphql/src/decorators/graphql-resolve-info.utils.ts:12]: Maybe just rename to `fields`?

**Comment by TriPSs** [comment]
Example of what `@GraphQLResolveInfo` outputs

```json
{
        "name": "todoItem",
        "args": {
          "id": 1
        },
        "fieldsByTypeName": {
          "tags": {
            "name": "tags",
            "args": {
              "sorting": [
                {
                  "field": "id",
                  "direction": "ASC"
                }
              ]
            },
            "fieldsByTypeName": {
              "pageInfo": {
                "name": "pageInfo",
                "args": {},
                "fieldsByTypeName": {
                  "hasNextPage": {
                    "name": "hasNextPage",
                    "args": {},
                    "fieldsByTypeName": {}
                  },
                  "hasPreviousPage": {
                    "name": "hasPreviousPage",
                    "args": {},
                    "fieldsByTypeName": {}
                  },
                  "startCursor": {
                    "name": "startCursor",
                    "args": {},
                    "fieldsByTypeName": {}
                  },
                  "endCursor": {
                    "name": "endCursor",
                    "args": {},
                    "fieldsByTypeName": {}
                  }
                }
              },
              "edges": {
                "name": "edges",
                "args": {},
                "fieldsByTypeName": {
                  "node": {
                    "name": "node",
                    "args": {},
                    "fieldsByTypeName": {
                      "id": {
                        "name": "id",
                        "args": {},
                        "fieldsByTypeName": {}
                      },
                      "name": {
                        "name": "name",
                        "args": {},
                        "fieldsByTypeName": {}
                      }
                    }
                  },
                  "cursor": {
                    "name": "cursor",
                    "args": {},
                    "fieldsByTypeName": {}
                  }
                }
              }
            }
          }
        }
      }
```

File: packages/query-graphql/src/decorators/graphql-resolve-info.utils.ts (hunk 2 above)
Comment (mrvinceit) [reply packages/query-graphql/src/decorators/graphql-resolve-info.utils.ts:12]: Agreed `fields` is better 😄

**Comment by mrvinceit** [comment]
I think the Connection object output should be filtered out and that can be done via a util function I suppose, but instead, I would move the fields found in the edges->node up under the tags->fields property

**Comment by TriPSs** [comment]
> I think the Connection object output should be filtered out and that can be done via a util function I suppose, but instead, I would move the fields found in the edges->node up under the tags->fields property

Agree, I have checked it a bit and that should be the [case now](https://github.com/TriPSs/nestjs-query/pull/70/files#diff-a8c4f481736b2c380dd5f16a492f5efd1da1e0e7a5f9f82d3408f1a1de71479fR96).

File: package.json [hunk 3]
```diff
@@ -121,5 +121,6 @@
     "tsconfig-paths": "3.14.1",
     "typescript": "4.9.4"
   },
-  "packageManager": "yarn@3.2.1"
+  "packageManager": "yarn@3.2.1",
+  "version": "2.2.0"
```
Comment (GP4cK) [inline package.json]: @TriPSs version is specified twice. Here and on line 3

**Pull request merged**